modules/feature-management/src/Volo.Abp.FeatureManagement.Application/Volo/Abp/FeatureManagement/FeatureAppService.cs

```diff
@@ -93,12 +102,22 @@ public virtual async Task UpdateAsync([NotNull] string providerName, [NotNull] s
             }
         }
 
-        protected virtual async Task CheckProviderPolicy(string providerName)
+        protected virtual async Task CheckProviderPolicy(string providerName, string providerKey)
         {
-            var policyName = Options.ProviderPolicies.GetOrDefault(providerName);
-            if (policyName.IsNullOrEmpty())
+            string policyName;
+            if (providerName == TenantFeatureValueProvider.ProviderName)
+            {
+                policyName = CurrentTenant.Id == null && providerKey == null ?
+                    "FeatureManagement.ManageHostFeatures" :
+                    "AbpTenantManagement.Tenants.ManageFeatures";
```

Comment:
SaaS module is also using here and the permission name is not `AbpTenantManagement.Tenants.ManageFeatures` in this case. See https://github.com/volosoft/volo/blob/dev/abp/saas/src/Volo.Saas.Host.Application.Contracts/Volo/Saas/Host/SaasHostPermissions.cs#L15

You only use `FeatureManagement.ManageHostFeatures` permission name for `providerName == TenantFeatureValueProvider.ProviderName && CurrentTenant.Id == null && providerKey == null`, but use from the Options.ProviderPolicies otherwise.